static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from "react";
+import {
+  Button,
+  ActionButton,
+  Input,
+  Select,
+  ValueOf,
+} from "@canonical/react-components";
+import { Formik, Form, Field } from "formik";
+
+import { HandleNewUserSubmit, UserRole } from "../../types";
+import { userRoleOptions } from "../../constants";
+
+interface Values {
+  email: string;
+  role: UserRole;
+  name: string;
+}
+
+const validateRequired = (value: string): string | undefined =>
+  !value ? "This field is required." : undefined;
+
+const validateEmail = (value: string): string | undefined =>
+  validateRequired(value) ||
+  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)
+    ? "Must be a valid email."
+    : undefined;
+
+const errorMessages = {
+  email_already_exists: "Account with this email address exists.",
+  default: "An unknown error has occurred.",
+} as const;
```

Comment:
What does `as const` do here?

Reply:
Converts the entire object to be type literals giving stricter type checking.

https://twitter.com/peter_makowski/status/1370104697634312195